pkgs/desktops/cinnamon/cjs/default.nix

```diff
@@ -0,0 +1,33 @@
+{ autoconf-archive, autoreconfHook, dbus_glib, fetchFromGitHub, gobjectIntrospection, pkgconfig, spidermonkey_52, stdenv, wrapGAppsHook }:
+
+stdenv.mkDerivation rec {
+  pname = "cjs";
+  version = "4.2.0";
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = "cjs";
+    rev = "${version}";
+    sha256 = "1yvzvfm1z0d8ca9vk9s0gbsir0ln7mcmlbczf0hh8vzpsg7m1zk5";
+  };
+
+  buildInputs = [ autoconf-archive dbus_glib gobjectIntrospection pkgconfig spidermonkey_52 ];
```

Comment:
It seems it may be possible for this to build with a later spidermonkey version
* https://github.com/linuxmint/cjs/blob/fe3b1ec434700f45bf75615285a18decfa3aca1a/NEWS#L360